tchannel/testing/vcr/cassette.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import absolute_import
+
+from itertools import chain
+from collections import deque
+
+from . import types
+from . import yaml
+from .exceptions import RequestNotFoundError, UnsupportedVersionError
+
+
+# Version of the storage format.
+VERSION = 1
+
+
+class Cassette(object):
+    """Represents a series of recorded interactions."""
+
+    def __init__(self, path):
+        """Initialize a new cassette.
+
+        :param path:
+            File path at which this cassette will be stored.
+        """
+        self.path = path
+
+        # Number of times interactions from this cassette have been played.
+        self.play_count = 0
+
+        self._available = deque()
+        self._played = deque()
+        self._recorded = deque()
+
+        self._load()
+
+    @property
+    def data(self):
+        """Get all known data for this cassette."""
+
+        return deque(
+            chain(self._played, self._available, self._recorded)
+        )
+
+    def _load(self):
+        try:
+            with open(self.path, 'r') as f:
```

Comment:
No. Why would we do that? That completely defeats the point of using a vcr library. You *want* to check the YAML files into your source repo along with your tests so that when you re-run the tests, your tests don't end up making another actual request to the dependent service.